testsuite/tests/lib-fun/protect.ml

```diff
@@ -0,0 +1,18 @@
+(* TEST
```

Comment:
Since the name of the function will likely change you could move the tests in `test.ml` itself.